OceanLab/eof.py

```diff
@@ -121,3 +121,162 @@ def my_eof_interp(M,nmodes,errmin=1e-15,repmax=None):
     vi = (M.T+Mmean).T
 
     return vi
+
+#=========================================
+# PERFORM COMPLEX EOF
+#=========================================
+from scipy.signal import hilbert
+import scipy.linalg as la
+from dask import delayed
+from scipy.signal import hilbert
+import xarray as xr
+
+def ceof(lon, lat, data, nkp = 10):
+    ''' Complex (Hilbert) EOF
+    First written in MATLAB and found in Prof. Daniel J. Vimont webpage 
+    (https://www.aos.wisc.edu/~dvimont/matlab/Stat_Tools/complex_eof.html)
+    
+    Translated to Python by Felipe Vilela da Silva @ IMAS, UTas, AU on 15/Mar/2021
+    ==============================================================================
+    Inputs:
+    -> lon: longitudes (array)
+    -> lat: latitude (array)
+    -> data: original data set [time, lat, lon]
+      -> Note: the mean field in each coordinate is subtracted within the function
+      -> Thus, do not remove the time-mean before inputing in the fuction.
+    -> nkp:  number of modes to output (default = 10)
+
+    ==============================================================================
+    Outputs in a DataArray:
+    -> per:   percent variance explained (real eigenvalues)
+    -> modes:  First nkp Complex Loadings or eigenvectors [lat, lon, nkp]
+    -> sp_amp:   Spatial amplitude [lat, lon, nkp]
+    -> sp_phase: Spatial phase [lat, lon, nkp]
+    -> pcs:   First nkp Complex Principal Components or amplitudes [time, nkp]
+    -> t_amp:    Temporal amplitude [time, nkp]
+    -> t_phase:  Temporal phase [time, nkp]    
+    ==============================================================================
```

Comment:
Please, remove all authorship comments (except for previous credits or paper citations). All contributors from OceanLab Team will be included in a single "contributors" file

Reply:
Done